tests/framework/framework/framework.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file was originally copied from https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go
+
+package framework
+
+import (
+	"fmt"
+	"math/rand"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/util"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"k8s.io/client-go/rest"
+
+	. "github.com/onsi/ginkgo"
+)
+
+// Framework supports common operations used by e2e tests; it will keep a client & clean the cluster for you.
+type Framework struct {
+	BaseName string
+
+	UniqueName string
```

Comment:
No, you didn't miss anything. Currently it is never used. I think that it could be useful for future improvements (i.g. create test specific labels). I can delete it if you think that is better to "add it when we need it"